### ISSUE TITLE
NullInjectorError: No provider for CookieService in Angular Service

### ISSUE THREAD

**Issue by chinggg** [salemdar/ngx-cookie#294]
**Describe the bug**
I want to migrate from `ngx-cookie-service` to `ngx-cookie`, and it seems that the two packages have much in common in grammar, so I can just use `grep` and `sed` to replace `set()` with `put()`, along with other small modifications. It works well for all Angular Components except one Angular Service. When I run `npm test`, it still throws out errors at `LocalBackupService` saying `NullInjectorError: No provider for CookieService! `, which have disappeared for all components after I `import { CookieModule }` and add `CookieModule.forRoot()` in `*.component.spec.ts`. Even after I add `CookieModule.forChild()`, these errors are still there.

**To Reproduce**
1. git clone `juice-shop`
2. do modifications like above
3. run `npm test`

**Expected behavior**
Pass all tests and no errors saying `No provider for CookieService`

**Stacktrace**
I am not familiar with NodeJS so I cannot figure out it. I guess there is a difference between `Component` and `Service`.

**Environment:**
 - Angular version [e.g. 11.2.3]
 - Browser [Chromium 89]
 - Version [e.g. 5.0.2]
 - OS: [e.g. Arch Linux]

**Additional context**
Pattern of errors
```
NullInjectorError: R3InjectorError(DynamicTestModule)[LocalBackupService -> CookieService -> CookieService]: 
	  NullInjectorError: No provider for CookieService!
	error properties: Object({ ngTempTokenPath: null, ngTokenPath: [ 'LocalBackupService', 'CookieService', 'CookieService' ] })
	NullInjectorError: R3InjectorError(DynamicTestModule)[LocalBackupService -> CookieService -> CookieService]: 
	  NullInjectorError: No provider for CookieService!
	    at NullInjector.get (node_modules/@angular/core/fesm2015/core.js:11045:1)
	    at R3Injector.get (node_modules/@angular/core/fesm2015/core.js:11212:1)
	    at R3Injector.get (node_modules/@angular/core/fesm2015/core.js:11212:1)
	    at injectInjectorOnly (node_modules/@angular/core/fesm2015/core.js:4723:1)
	    at ɵɵinject (node_modules/@angular/core/fesm2015/core.js:4727:1)
	    at Object.LocalBackupService_Factory [as factory] (ng:///LocalBackupService/ɵfac.js:4:46)
	    at R3Injector.hydrate (node_modules/@angular/core/fesm2015/core.js:11380:1)
	    at R3Injector.get (node_modules/@angular/core/fesm2015/core.js:11201:1)
	    at NgModuleRef$1.get (node_modules/@angular/core/fesm2015/core.js:25276:1)
	    at Object.get (node_modules/@angular/core/fesm2015/core.js:24990:1)
```

**Comment by salemdar** [comment]
You can import the module, that is the quick solution. But a better way would be refactoring your tests and mocking the `CookieService`:

```typescript
describe('MyService', () => {
  // ...
  let service: MyService;
  let cookieServiceSpy: SpyObj<CookieService>;

  beforeEach(waitForAsync(() => {
    // ...
    cookieServiceSpy = jasmine.createSpyObj<CookieService>('CookieService', ['put', 'get']);
    TestBed.configureTestingModule({
      providers: [
        // ...
        {provide: CookieService, useValue: cookieServiceSpy}
      ]
    })
      .compileComponents()
  }));

  it('should set some cookies in some case', () => {
    service.methodThatLeadsToSavingACookie();
    expect(cookieServiceSpy.put).toHaveBeenCalled();
  });
});
```

**Comment by chinggg** [comment]
@salemdar Thanks! I found the errors are caused by lack of provider declaration. I added these lines and the errors disappeared.
```diff
+ let localBackupService: any
      providers: [
+        { provide: LocalBackupService, useValue: localBackupService },
      ]
```

Btw, I want to know whether `ngx-cookie` will leave the `SameSite` attribute of new cookies unspecified. Since `ngx-cookie-service` switched the default for new cookies to `SameSite=Lax`, see stevermeister/ngx-cookie-service@0cb4839.

That is why I have been trying to migrate from `ngx-cookie-service` to `ngx-cookie`.
